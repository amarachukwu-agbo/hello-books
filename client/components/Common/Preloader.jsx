import React from 'react';

/**
 * @description stateless component for loader
 * that is displayed while data is loading
 *
 * @returns {Node} - react node containing the Preloader component
 */
const Preloader = () => (
    <div className="preloader-wrapper medium active">
    <div className="spinner-layer spinner-blue-only">
      <div className="circle-clipper left">
        <div className="circle"></div>
      </div><div className="gap-patch">
        <div className="circle"></div>
      </div><div className="circle-clipper right">
        <div className="circle"></div>
      </div>
    </div>
  </div>
);

export default Preloader;
