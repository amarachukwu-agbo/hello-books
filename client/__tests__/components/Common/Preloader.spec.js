import React from 'react';
import { shallow } from 'enzyme';
import Preloader from '../../../components/Common/Preloader';

describe('<Preloader>', () => {
  it('should render the preloader component correctly', () => {
    const tree = shallow(<Preloader />);
    expect(tree).toMatchSnapshot();
  });
});
