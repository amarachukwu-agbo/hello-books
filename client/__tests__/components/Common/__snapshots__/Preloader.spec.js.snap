// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Preloader> should render the preloader component correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Preloader />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="spinner-layer spinner-blue-only"
      >
        <div
          className="circle-clipper left"
        >
          <div
            className="circle"
          />
        </div>
        <div
          className="gap-patch"
        >
          <div
            className="circle"
          />
        </div>
        <div
          className="circle-clipper right"
        >
          <div
            className="circle"
          />
        </div>
      </div>,
      "className": "preloader-wrapper medium active",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="circle-clipper left"
          >
            <div
              className="circle"
            />
          </div>,
          <div
            className="gap-patch"
          >
            <div
              className="circle"
            />
          </div>,
          <div
            className="circle-clipper right"
          >
            <div
              className="circle"
            />
          </div>,
        ],
        "className": "spinner-layer spinner-blue-only",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="circle"
            />,
            "className": "circle-clipper left",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "circle",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="circle"
            />,
            "className": "gap-patch",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "circle",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="circle"
            />,
            "className": "circle-clipper right",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "circle",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="spinner-layer spinner-blue-only"
        >
          <div
            className="circle-clipper left"
          >
            <div
              className="circle"
            />
          </div>
          <div
            className="gap-patch"
          >
            <div
              className="circle"
            />
          </div>
          <div
            className="circle-clipper right"
          >
            <div
              className="circle"
            />
          </div>
        </div>,
        "className": "preloader-wrapper medium active",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="circle-clipper left"
            >
              <div
                className="circle"
              />
            </div>,
            <div
              className="gap-patch"
            >
              <div
                className="circle"
              />
            </div>,
            <div
              className="circle-clipper right"
            >
              <div
                className="circle"
              />
            </div>,
          ],
          "className": "spinner-layer spinner-blue-only",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="circle"
              />,
              "className": "circle-clipper left",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "circle",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="circle"
              />,
              "className": "gap-patch",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "circle",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="circle"
              />,
              "className": "circle-clipper right",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "circle",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
